OutSystems/outsystems-datagrid
# ROU-4702: fixing apis FilterByCondition and FilterByValue when filtering by NullDate

This PR is fixing the APIs for the filters when empty of NullDate is passed.

### What was happening
* When the developer wanted to filter by nulldate or empty string (meaning that the records would have no date coming from the data base. The filter would miss behave and always show no results.
* If the end user would perform the same operation manually, the grid would should the correct values.
* The problem was that the data was not being "treated" before being applied to the filter.

### What was done
* Created two new global constants with the `NullDate` values of the platform.
* In the class `ColumnFilter` added a new private method `_isValidDate` that checks if the input is a valid date (`date not int [undefined, null, "", 1900-01-01, 1900-01-01Z00:00:00`).
* Changed `_getFilterConditionValue`, where the value for the filter is set as `null` in case the `_isValidDate` returns `false`.
* Changed `byValue`, to also consider the result of `_isValidDate` and in case it returns `false`, the value `''` is set.

### Test Steps
1. Go to the test page
2. Choose the value to for the filter
3. Apply the filter (value or condition)
4. Filter will be applied to the Date column, and the number of results should be reflected (see gif)


### Screenshots
![datagrid-filter](https://github.com/OutSystems/outsystems-datagrid/assets/10534623/c283b00f-e0c7-4000-b51f-441f78691e89)


### Checklist
